lib/resources/datav2/rest_v2.ts

```diff
@@ -539,3 +566,78 @@ export async function getCryptoSnapshot(
 
   return AlpacaCryptoSnapshot(resp.data);
 }
+
+export enum Sort {
+  ASC = "asc",
+  DESC = "desc",
+}
+
+export interface GetNewsParams {
+  // Symbols filters the news to the related symbols.
+  // If empty or nil, all articles will be returned.
+  symbols: Array<string>;
+  // Start is the inclusive beginning of the interval
+  start?: string;
+  // End is the inclusive end of the interval
+  end?: string;
+  // Sort sets the sort order of the results. Sorting will be done by the UpdatedAt field.
+  sort?: Sort;
+  // IncludeContent tells the server to include the article content in the response.
+  includeContent?: boolean;
+  // ExcludeContentless tells the server to exclude articles that has no content.
+  excludeContentless?: boolean;
+  // TotalLimit is the limit of the total number of the returned news.
+  totalLimit?: number;
+  // PageLimit is the pagination size. If empty, the default page size will be used.
+  pageLimit?: number;
+}
+
+function getNewsParams(options: GetNewsParams): any {
+  const query = {} as any;
+  query.symbols =
+    options.symbols?.length > 0 ? options.symbols.join(",") : null;
+  query.start = options.start ? options.start : null;
+  query.end = options.end ? options.end : null;
+  query.sort = options.sort ? options.sort : null;
+  query.includeContent = options.includeContent ? options.includeContent : null;
+  query.excludeContentless = options.excludeContentless
+    ? options.excludeContentless
+    : null;
+  return query;
+}
+
+export async function getNews(
+  options: GetNewsParams,
+  config: any
+): Promise<AlpacaNews[]> {
+  let pageToken: string | null = null;
+  let received = 0;
+  const pageLimit = options.pageLimit
+    ? Math.min(options.pageLimit, V2_NEWS_MAX_LIMIT)
+    : V2_NEWS_MAX_LIMIT;
+  const result: AlpacaNews[] = [];
+  const totalLimit = options.totalLimit
+    ? options.totalLimit
+    : V2_NEWS_MAX_LIMIT;
+  delete options.pageLimit;
+  const params = getNewsParams(options);
+  for (;;) {
+    const limit = getQueryLimit(totalLimit, pageLimit, received);
+    if (limit < 1) {
+      break;
+    }
+
+    const resp: AxiosResponse<any> = await dataV2HttpRequest(
+      "/v1beta1/news",
+      { ...params, limit: limit, page_token: pageToken },
+      config
+    );
+    resp.data.news.map((n: RawAlpacaNews) => result.push(AlpacaNews(n)));
```

Comment:
map is used as a pure function that has no sideeffect and the return value is used . For this usecase foreach is the better approach